services/payments/src/simcore_service_payments/api/rpc/_payments_methods.py

```diff
@@ -0,0 +1,267 @@
+from fastapi import FastAPI
+from models_library.api_schemas_webserver.wallets import (
+    PaymentMethodGet,
+    PaymentMethodID,
+    PaymentMethodInitiated,
+    WalletPaymentInitiated,
+)
+from models_library.basic_types import IDStr
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from pydantic import EmailStr
+from servicelib.logging_utils import get_log_record_extra, log_context
+from servicelib.rabbitmq import RPCRouter
+from simcore_postgres_database.models.payments_methods import InitPromptAckFlowState
+from simcore_service_payments.db.payments_methods_repo import PaymentsMethodsRepo
+from simcore_service_payments.models.db import PaymentsMethodsDB
+from simcore_service_payments.models.payments_gateway import (
+    GetPaymentMethod,
+    InitPayment,
+    InitPaymentMethod,
+)
+
+from ..._constants import PAG, PGDB
+from ...db.payments_transactions_repo import PaymentsTransactionsRepo
+from ...services.payments_gateway import PaymentsGatewayApi
+
+_logger = logging.getLogger(__name__)
+
+
+router = RPCRouter()
+
+
+@router.expose()
+async def init_creation_of_payment_method(
+    app: FastAPI,
+    *,
+    wallet_id: WalletID,
+    wallet_name: IDStr,
+    user_id: UserID,
+    user_name: IDStr,
+    user_email: EmailStr,
+) -> PaymentMethodInitiated:
+    initiated_at = arrow.utcnow().datetime
+
+    # Payment-Gateway
+    with log_context(
+        _logger,
+        logging.INFO,
+        "%s: Init payment-method %s in payments-gateway",
+        PAG,
+        f"{wallet_id=}",
+        extra=get_log_record_extra(user_id=user_id),
+    ):
+        gateway = PaymentsGatewayApi.get_from_app_state(app)
+
+        init = await gateway.init_payment_method(
+            InitPaymentMethod(
+                user_name=user_name,
+                user_email=user_email,
+                wallet_name=wallet_name,
+            )
+        )
+
+        form_link = gateway.get_form_payment_method_url(init.payment_method_id)
+
+    # Database
+    with log_context(
+        _logger,
+        logging.INFO,
+        "%s: Annotate INIT payment-method %s in db",
+        PGDB,
+        f"{init.payment_method_id=}",
+        extra=get_log_record_extra(user_id=user_id),
+    ):
+        repo = PaymentsMethodsRepo(db_engine=app.state.engine)
+        payment_method_id = await repo.insert_init_payment_method(
+            init.payment_method_id,
+            user_id=user_id,
+            wallet_id=wallet_id,
+            initiated_at=initiated_at,
+        )
+        assert payment_method_id == init.payment_method_id  # nosec
+
+    return PaymentMethodInitiated(
+        wallet_id=wallet_id,
+        payment_method_id=payment_method_id,
+        payment_method_form_url=f"{form_link}",
+    )
+
+
+@router.expose()
+async def cancel_creation_of_payment_method(
+    app: FastAPI,
+    *,
+    payment_method_id: PaymentMethodID,
+    user_id: UserID,
+    wallet_id: WalletID,
+) -> None:
+    # TODO: with db transaction
```

Comment:
was this forgotten as well?

Reply:
Very very sorry ... i should not PR at night!